tools/logger/convert.c

```diff
@@ -131,14 +131,53 @@ static const char *format_uid(uint32_t uid_ptr, int use_colors, bool be, bool up
 	return str;
 }
 
+/* fmt should point '%pUx`, return pointer to UUID string or zero */
+static const char *asprintf_uuid(const char *fmt, uint32_t uuid_key, bool use_colors, int *fmt_len)
+{
+	const char *fmt_end = fmt + strlen(fmt);
+	int be, upper;
+	int len = 4; /* assure full formating, with x */
+
+	if (fmt + 2 >= fmt_end || fmt[1] != 'p' || fmt[2] != 'U') {
+		*fmt_len = 0;
+		return 0;
```

Comment:
Shouldn't you return NULL instead of 0 here? The return type is `const char *`.

Reply:
replaced